app/src/main/java/com/nio/TcpDataFlowExample.java

```diff
@@ -1,117 +1,236 @@
-
-						clients.put(server.socket().getLocalPort(), client);
-
-						client.register(selector, SelectionKey.OP_READ);
-					}
-
-					if (selectionKey.isReadable()) {
-						System.out.println("Handle READ event");
-						final SocketChannel client = (SocketChannel) selectionKey.channel();
-						final int read = client.read(buffer);
-
-						if (read == -1) {
-							client.close();
-							client.keyFor(selector).cancel();
-							System.out.printf("The connection was closed: %s%n", client);
-							return;
-						}
-
-						buffer.flip();
-						switch (client.socket().getLocalPort()) {
-						case 5555:
-							for (int i = 0; i < buffer.limit(); i++) {
-								buffer.put(i, (byte) Character.toUpperCase(buffer.get(i)));
-							}
-							client.write(buffer);
-							break;
-						case 4444:
-							final String command = StandardCharsets.UTF_8.decode(buffer).toString();
-							final String stopCommand = "stop-read";
-							if (stopCommand.equals(command.trim().toLowerCase())) {
-								System.out.println("Handle stop-read");
-								registerEvent(selector, clients, 0);
-								break;
-							}
-							final String startCommand = "start-read";
-							if (startCommand.equals(command.trim().toLowerCase())) {
-								System.out.println("Handle start-read");
-								registerEvent(selector, clients, SelectionKey.OP_READ);
-								break;
-							}
-
-							final byte[] unknownCommand = String.format("Supported commands:%n%s%n%s%n", stopCommand, startCommand)
-									.getBytes(StandardCharsets.UTF_8);
-							client.write(ByteBuffer.wrap(unknownCommand));
-							break;
-						}
-
-						buffer.clear();
-					}
-					iterator.remove();
-				} catch (Exception e) {
-					e.printStackTrace();
-				}
-			});
-		}
-	}
-
-	private static void registerEvent(final Selector selector, final Multimap<Integer, SocketChannel> clients, final int op) {
-		clients.get(5555).forEach(c -> {
-			try {
-				c.register(selector, op);
-			} catch (Exception e) {
-				e.printStackTrace();
-			}
-		});
-	}
+    private TcpDataFlowExample() {
+    }
+
+    public static void main(final String... args) throws Exception {
+        Properties p = readProperties();
+        String hostname = Config.HOSTNAME.get(p);
+        int controlChannelPort = Integer.parseInt(Config.CONTROL_CHANNEL_PORT.get(p));
+        int dataChannelPort = Integer.parseInt(Config.DATA_CHANNEL_PORT.get(p));
+
+        System.out.printf("Tcp Data Flow Example started at %s:[%d, %d]%n", hostname, controlChannelPort, dataChannelPort);
+
+        final Selector selector = Selector.open();
+        final List<Channel> channels = Arrays.asList(
+                new ControlChannel(hostname, controlChannelPort, selector),
+                new DataChannel(hostname, dataChannelPort, selector)
+        );
+        final ByteBuffer buffer = ByteBuffer.allocate(65535);
+        final Collection<SocketChannel> clients = new ArrayList<>();
+
+        while (!Thread.currentThread().isInterrupted()) {
```

Comment:
There is no way to use the service as API. The main method is blocked in the main thread.